concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -365,10 +338,39 @@ loadSkovPersistentData rp _treeStateDirectory _genesisData pbsc atiContext = do
     makeBlockPointer StoredBlock{..} = do
       bstate <- runReaderT (PBS.runPersistentBlockStateMonad (loadBlockState (blockStateHash sbBlock) sbState)) pbsc
       makeBlockPointerFromPersistentBlock sbBlock bstate defaultValue sbInfo
-    makeBlockPointerCached :: StoredBlock pv (TS.BlockStatePointer (PBS.PersistentBlockState pv)) -> IO (PersistentBlockPointer pv (ATIValues ati) (PBS.HashedPersistentBlockState pv))
-    makeBlockPointerCached StoredBlock{..} = do
-      bstate <- runReaderT (PBS.runPersistentBlockStateMonad (cacheBlockState =<< loadBlockState (blockStateHash sbBlock) sbState)) pbsc
-      makeBlockPointerFromPersistentBlock sbBlock bstate defaultValue sbInfo
+
+activateSkovPersistentData :: forall ati pv. (IsProtocolVersion pv, CanExtend (ATIValues ati))
+                           => PBS.PersistentBlockStateContext
+                           -> SkovPersistentData pv ati (PBS.HashedPersistentBlockState pv)
+                           -> LogIO (SkovPersistentData pv ati (PBS.HashedPersistentBlockState pv))
+activateSkovPersistentData pbsc uninitState = do
+  cachedLastFinalized <- liftIO (makeBlockPointerCached (_lastFinalized uninitState))
+  -- We need to establish is the transaction table invariants.
```

Comment:
I see that this comment came unchanged from the body of `loadSkovPersistentData` after `getTransactionTable` was moved. It'd be better to list the invariants that `activateSkovPersistentData` checks upfront, like it's done in the comment documenting `loadSkovPersistentData`.

There's, perhaps, only one invariant that's checked here ("every account in the account lists is stored in the database"), but still the function description will be easier to read if it says explicitly what needs to be true to run this function.

There's also no need to say anything about next available nonces when talking about this invariant, I believe — that's not what's being checked.

Reply:
I don't know exactly what you are referring to. This function is establishing the invariants. It is not checking them.

And the state invariant it is establishing is that the pending table must record the correct "next nonce" for each account.